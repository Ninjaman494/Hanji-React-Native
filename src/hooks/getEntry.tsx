import { gql, QueryResult, useQuery } from "@apollo/client";

const ENTRY = gql`
  query EntryQuery($id: ID!) {
    entry(id: $id) {
      id
      term
      pos
      definitions
      antonyms
      synonyms
      examples {
        sentence
        translation
      }
      regular
      note
    }
  }
`;

type Entry = {
  id: string;
  term: string;
  pos: string;
  definitions: string[];
  antonyms?: string[];
  synonyms?: string[];
  examples?: {
    sentence: string;
    translation: string;
  }[];
  regular?: boolean;
  note?: string;
};

type GetEntryResponse = {
  entry: Entry;
};

export default function getEntry(id: string): QueryResult<GetEntryResponse> {
  return useQuery<GetEntryResponse>(ENTRY, { variables: { id: id } });
}
