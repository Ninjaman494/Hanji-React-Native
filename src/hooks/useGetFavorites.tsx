import { useEffect, useState } from "react";
import AsyncStorage from "@react-native-async-storage/async-storage";

export type Favorite = {
  name: string;
  conjugationName: string;
  honorific: boolean;
};

export const FAVORITES_KEY = "@favorites_Key";

const useGetFavorites = () => {
  const [favorites, setFavorites] = useState<Favorite[] | null>(null);

  useEffect(() => {
    (async () => {
      try {
        const value = await AsyncStorage.getItem(FAVORITES_KEY);
        if (value !== null) {
          setFavorites(JSON.parse(value));
        } else {
          setFavorites([]);
        }
      } catch (e) {
        console.log(e);
      }
    })();
  }, [setFavorites]);

  return favorites;
};

export default useGetFavorites;
