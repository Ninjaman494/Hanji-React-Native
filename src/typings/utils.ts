export type FormikContext = Record<
  string,
  string | number | boolean | Record<string, unknown>
>;
