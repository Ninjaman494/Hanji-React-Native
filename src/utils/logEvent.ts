import analytics, {
  FirebaseAnalyticsTypes,
} from "@react-native-firebase/analytics";

export enum LOG_EVENT {
  // Predefined events
  SEARCH = "search",
  SELECT_CONTENT = "select_content",
  SCREEN_VIEW = "screen_view",
  SELECT_ITEM = "select_item",
  PURCHASE = "purchase",
  // Custom events
  SELECT_CONJUGATION = "select_conjugation",
  ADD_FAVORITE = "add_favorite",
  SELECT_FAV = "select_fav",
  REPORT_BUG = "report_bug",
  MAKE_SUGGESTION = "make_suggestion",
  SUBMIT_SURVEY = "submit_survey",
}

type LogEventVars =
  | {
      type: LOG_EVENT.SEARCH;
      params: FirebaseAnalyticsTypes.SearchEventParameters;
    }
  | {
      type: LOG_EVENT.SELECT_CONTENT;
      params: FirebaseAnalyticsTypes.SelectContentEventParameters;
    }
  | {
      type: LOG_EVENT.PURCHASE;
      params: FirebaseAnalyticsTypes.PurchaseEventParameters;
    }
  | {
      type: LOG_EVENT.SCREEN_VIEW;
      params: FirebaseAnalyticsTypes.ScreenViewParameters;
    }
  | {
      type: LOG_EVENT.SELECT_ITEM;
      params: FirebaseAnalyticsTypes.SelectItemEventParameters;
    }
  | {
      type:
        | LOG_EVENT.SELECT_CONJUGATION
        | LOG_EVENT.ADD_FAVORITE
        | LOG_EVENT.SELECT_FAV
        | LOG_EVENT.REPORT_BUG
        | LOG_EVENT.MAKE_SUGGESTION
        | LOG_EVENT.SUBMIT_SURVEY;
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      params: Record<string, any>;
    };

const logEvent = async (vars: LogEventVars): Promise<void> => {
  switch (vars.type) {
    case LOG_EVENT.SEARCH:
      await analytics().logSearch(vars.params);
      break;
    case LOG_EVENT.SELECT_CONTENT:
      await analytics().logSelectContent(vars.params);
      break;
    case LOG_EVENT.PURCHASE:
      await analytics().logPurchase(vars.params);
      break;
    case LOG_EVENT.SCREEN_VIEW:
      await analytics().logScreenView(vars.params);
      break;
    case LOG_EVENT.SELECT_ITEM:
      await analytics().logSelectItem(vars.params);
      break;
    default:
      await analytics().logEvent(vars.type, vars.params);
  }
};

export default logEvent;
