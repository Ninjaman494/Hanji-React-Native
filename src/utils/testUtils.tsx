import {
  render,
  RenderAPI,
  RenderOptions,
} from "@testing-library/react-native";
import React, { FC, PropsWithChildren, ReactElement } from "react";
import { Provider as PaperProvider } from "react-native-paper";
import theme from "theme";

const Providers: FC<PropsWithChildren> = ({ children }) => {
  return <PaperProvider theme={theme}>{children}</PaperProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderAPI => render(ui, { wrapper: Providers, ...options });

export * from "@testing-library/react-native";
export { customRender as render };
