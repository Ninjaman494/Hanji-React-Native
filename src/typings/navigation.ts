import { StackNavigationProp, StackScreenProps } from "@react-navigation/stack";
import { Conjugation } from "hooks/useConjugations";

export enum PageName {
  MAIN = "Main",
  SEARCH = "Search",
  DISPLAY = "Display",
  CONJUGATIONS = "Conjugations",
  CONJINFO = "ConjInfo",
  SETTINGS = "Settings",
  FAVORITES = "Favorites",
  ACKNOWLEDGEMENTS = "Acknowledgements",
  BUGREPORT = "BugReport",
  CONJUGATOR = "Conjugator",
}

export type StackParamList = {
  [PageName.MAIN]: undefined;
  [PageName.SEARCH]: { query: string };
  [PageName.DISPLAY]: { entryId: string; noAnimate?: boolean };
  [PageName.CONJUGATIONS]: { stem: string; isAdj: boolean; honorific: boolean };
  [PageName.CONJINFO]: { conjugation: Conjugation };
  [PageName.SETTINGS]: undefined;
  [PageName.FAVORITES]: undefined;
  [PageName.ACKNOWLEDGEMENTS]: undefined;
  [PageName.BUGREPORT]: { screenshot: string };
  [PageName.CONJUGATOR]: { term: string };
};

export type NavigationProps = StackNavigationProp<StackParamList>;

export type ScreenProps<RouteName extends keyof StackParamList> =
  StackScreenProps<StackParamList, RouteName>;
