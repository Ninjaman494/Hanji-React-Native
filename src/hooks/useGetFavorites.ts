import { useCallback, useEffect, useState } from "react";
import { FAVORITES_KEY, getAsyncStorage } from "utils/asyncStorageHelper";
import { ConjugationName } from "utils/conjugationTypes";

export type Favorite = {
  name: string;
  conjugationName: ConjugationName;
  honorific: boolean;
};

export interface GetFavoritesResponse {
  favorites: Favorite[] | null;
  loading: boolean;
  error: Error | null;
  refetch: () => Promise<void>;
}

const useGetFavorites = (): GetFavoritesResponse => {
  const [favorites, setFavorites] = useState<Favorite[] | null>(null);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState<Error | null>(null);

  const fetchFavorites = useCallback(async () => {
    try {
      const value = await getAsyncStorage(FAVORITES_KEY, "object");
      setFavorites(value as Favorite[] | null);
    } catch (e) {
      setError(e as Error);
    } finally {
      setLoading(false);
    }
  }, [setFavorites, setLoading, setError]);

  useEffect(() => {
    fetchFavorites();
  }, [fetchFavorites]);

  return { favorites, loading, error, refetch: fetchFavorites };
};

export default useGetFavorites;
