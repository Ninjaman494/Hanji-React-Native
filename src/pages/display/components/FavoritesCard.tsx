import { useNavigation } from "@react-navigation/native";
import { BaseCard, BaseCardProps } from "components";
import { Conjugation } from "hooks/useConjugations";
import { Favorite } from "hooks/useGetFavorites";
import React from "react";
import { StyleSheet } from "react-native";
import { Col, Grid, Row } from "react-native-easy-grid";
import { Button, Card, Text, useTheme } from "react-native-paper";
import { StackNavigationProp } from "typings/navigation";

export type FavoritesCardProps = BaseCardProps & {
  favorites: Favorite[];
  conjugations?: Conjugation[];
  onPress: () => void;
};

const FavoritesCard: React.FC<FavoritesCardProps> = ({
  favorites,
  conjugations,
  onPress,
  ...rest
}) => {
  const { colors, padding, textSizes } = useTheme();
  const history = useNavigation<StackNavigationProp>();

  const style = StyleSheet.create({
    conjView: {
      paddingHorizontal: padding.horizontal,
    },
    rowView: {
      marginBottom: padding.vertical,
    },
    text: {
      fontSize: textSizes.regular,
    },
    divider: {
      fontSize: textSizes.regular,
      textAlign: "center",
    },
    actions: {
      justifyContent: "flex-end",
      paddingBottom: 0,
    },
  });

  const favoritesMap = conjugations?.length
    ? favorites.map((favorite) => {
        const conjugation = conjugations?.find(
          ({ name, honorific }) =>
            name === favorite.conjugationName &&
            honorific === favorite.honorific
        );
        return { name: favorite.name, conjugation: conjugation };
      })
    : [];

  return (
    <BaseCard {...rest}>
      <Grid style={style.conjView}>
        {favoritesMap.length > 0 ? (
          favoritesMap.map((favorite, index) => (
            <Row
              style={style.rowView}
              key={index}
              onPress={() =>
                history.push("ConjInfo", {
                  conjugation: favorite.conjugation as Conjugation,
                })
              }
            >
              <Col size={5}>
                <Text style={style.text}>{favorite.name}</Text>
              </Col>
              <Col>
                <Text style={style.divider}>:</Text>
              </Col>
              <Col size={5}>
                <Text style={style.text}>
                  {favorite.conjugation?.conjugation}
                </Text>
              </Col>
            </Row>
          ))
        ) : (
          <Text style={style.text}>
            You don't have any favorites. Click on favorites in settings to make
            some.
          </Text>
        )}
      </Grid>
      <Card.Actions style={style.actions}>
        <Button onPress={onPress} color={colors.accent}>
          See all
        </Button>
      </Card.Actions>
    </BaseCard>
  );
};

export default FavoritesCard;
