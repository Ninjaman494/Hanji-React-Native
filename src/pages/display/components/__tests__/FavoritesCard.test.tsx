import { useNavigation } from "@react-navigation/native";
import { Favorite } from "hooks/useGetFavorites";
import React from "react";
import "react-native";
import { NavigationProps } from "typings/navigation";
import {
  ConjugationName,
  ConjugationType,
  Formality,
  Tense,
} from "utils/conjugationTypes";
import logEvent, { LOG_EVENT } from "utils/logEvent";
import { fireEvent, render, waitFor } from "utils/testUtils";
import FavoritesCard from "../../components/FavoritesCard";

const { push } = useNavigation<NavigationProps>();

const favorites: Favorite[] = [
  {
    name: "Favorite 1",
    conjugationName: "declarative past informal high",
    honorific: false,
  },
  {
    name: "Favorite 2",
    conjugationName: "declarative present informal high",
    honorific: false,
  },
  {
    name: "Favorite 3",
    conjugationName: "declarative future informal high",
    honorific: false,
  },
];

const baseConjugation = {
  type: "type",
  tense: Tense.PRESENT,
  speechLevel: Formality.INFORMAL_HIGH,
  honorific: false,
  pronunciation: "pronunciation",
  romanization: "romanization",
  reasons: [],
};

const props = {
  favorites: favorites,
  conjugations: [
    {
      name: "declarative past informal high" as ConjugationName,
      conjugation: "conj 1",
      ...baseConjugation,
    },
    {
      name: "declarative present informal high" as ConjugationName,
      conjugation: "conj 2",
      ...baseConjugation,
    },
    {
      name: "declarative future informal high" as ConjugationName,
      conjugation: "conj 3",
      ...baseConjugation,
    },
  ],
  onPress: jest.fn(),
};

describe("FavoritesCard component", () => {
  it("displays favorites", () => {
    const component = render(<FavoritesCard {...props} />);
    favorites.forEach(({ name }) =>
      expect(component.getByText(name)).toBeTruthy()
    );
    props.conjugations.forEach(({ conjugation }) =>
      expect(component.getByText(conjugation)).toBeTruthy()
    );
  });

  it("displays empty state", () => {
    const component = render(<FavoritesCard {...props} favorites={[]} />);
    expect(
      component.getByText(
        "You don't have any favorites. Click on favorites in settings to make some."
      )
    ).toBeTruthy();
  });

  it("triggers onClick", async () => {
    const component = render(<FavoritesCard {...props} />);

    fireEvent.press(component.getByText("See all"));

    await waitFor(() => expect(props.onPress).toHaveBeenCalled());
  });

  it("redirects to ConjInfo page", async () => {
    const component = render(<FavoritesCard {...props} />);

    fireEvent.press(component.getByText("Favorite 1"));

    const conjugation = props.conjugations[0];
    await waitFor(() => {
      expect(logEvent).toHaveBeenCalledWith({
        type: LOG_EVENT.SELECT_FAV,
        params: {
          name: props.favorites[0].name,
          conjugation_name: conjugation?.name,
          conjugated: conjugation?.conjugation,
          honorific: conjugation?.honorific,
        },
      });
      expect(push).toHaveBeenCalledWith("ConjInfo", { conjugation });
    });
  });

  it("hides favorites with no conjugations", async () => {
    const customFavorites = [
      favorites[0],
      {
        name: "Favorite 2",
        conjugationName: ConjugationType.DETERMINER_PAST as ConjugationName,
        honorific: false,
      },
    ];

    const component = render(
      <FavoritesCard
        favorites={customFavorites}
        conjugations={[props.conjugations[0]]}
        onPress={jest.fn()}
      />
    );

    expect(component.getByText("Favorite 1")).toBeTruthy();
    expect(component.queryByText("Favorite 2")).toBeNull();
  });
});
