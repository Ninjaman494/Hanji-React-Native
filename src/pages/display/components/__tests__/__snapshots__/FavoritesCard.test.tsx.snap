// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesCard component displays favorites 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    nativeID="animatedComponent"
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "rgba(0, 0, 0, 0.12)",
        "borderRadius": 10,
        "elevation": 5,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 5,
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexGrow": 1,
          "flexShrink": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "paddingHorizontal": 16,
            },
            undefined,
            Object {
              "paddingBottom": 8,
              "paddingHorizontal": 0,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "paddingHorizontal": 16,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 8,
                "opacity": 1,
              }
            }
          >
            <View
              onPress={[Function]}
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "marginBottom": 8,
                }
              }
            >
              <View
                size={5}
                style={
                  Object {
                    "flex": 5,
                    "flexDirection": "column",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "fontSize": 20,
                      },
                    ]
                  }
                >
                  Favorite 1
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "fontSize": 20,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  :
                </Text>
              </View>
              <View
                size={5}
                style={
                  Object {
                    "flex": 5,
                    "flexDirection": "column",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "fontSize": 20,
                      },
                    ]
                  }
                >
                  conj 1
                </Text>
              </View>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 8,
                "opacity": 1,
              }
            }
          >
            <View
              onPress={[Function]}
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "marginBottom": 8,
                }
              }
            >
              <View
                size={5}
                style={
                  Object {
                    "flex": 5,
                    "flexDirection": "column",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "fontSize": 20,
                      },
                    ]
                  }
                >
                  Favorite 2
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "fontSize": 20,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  :
                </Text>
              </View>
              <View
                size={5}
                style={
                  Object {
                    "flex": 5,
                    "flexDirection": "column",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "fontSize": 20,
                      },
                    ]
                  }
                >
                  conj 2
                </Text>
              </View>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 8,
                "opacity": 1,
              }
            }
          >
            <View
              onPress={[Function]}
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "marginBottom": 8,
                }
              }
            >
              <View
                size={5}
                style={
                  Object {
                    "flex": 5,
                    "flexDirection": "column",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "fontSize": 20,
                      },
                    ]
                  }
                >
                  Favorite 3
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "fontSize": 20,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  :
                </Text>
              </View>
              <View
                size={5}
                style={
                  Object {
                    "flex": 5,
                    "flexDirection": "column",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "fontSize": 20,
                      },
                    ]
                  }
                >
                  conj 3
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "padding": 8,
              },
              Object {
                "justifyContent": "flex-end",
                "paddingBottom": 0,
              },
            ]
          }
        >
          <View
            collapsable={false}
            nativeID="animatedComponent"
            style={
              Object {
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderRadius": 10,
                "borderStyle": "solid",
                "borderWidth": 0,
                "elevation": 0,
                "minWidth": "auto",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 10,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "letterSpacing": 1,
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        Object {
                          "marginHorizontal": 8,
                        },
                        Object {
                          "textTransform": "uppercase",
                        },
                        Object {
                          "color": "#F44336",
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        Object {
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  See all
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
